src/display/network.js

```diff
@@ -351,51 +351,24 @@ function PDFNetworkStreamFullRequestReader(manager, options) {
 }
 
 PDFNetworkStreamFullRequestReader.prototype = {
-  _validateRangeRequestCapabilities: function
-      PDFNetworkStreamFullRequestReader_validateRangeRequestCapabilities() {
+  getResponseHeader(name) {
+    let fullRequestXhrId = this._fullRequestId;
+    let fullRequestXhr = this._manager.getRequestXhr(fullRequestXhrId);
 
-    if (this._disableRange) {
-      return false;
-    }
-
-    var networkManager = this._manager;
-    if (!networkManager.isHttp) {
-      return false;
-    }
-    var fullRequestXhrId = this._fullRequestId;
-    var fullRequestXhr = networkManager.getRequestXhr(fullRequestXhrId);
-    if (fullRequestXhr.getResponseHeader('Accept-Ranges') !== 'bytes') {
-      return false;
-    }
-
-    var contentEncoding =
-      fullRequestXhr.getResponseHeader('Content-Encoding') || 'identity';
-    if (contentEncoding !== 'identity') {
-      return false;
-    }
-
-    var length = fullRequestXhr.getResponseHeader('Content-Length');
-    length = parseInt(length, 10);
-    if (!isInt(length)) {
-      return false;
-    }
-
-    this._contentLength = length; // setting right content length
-
-    if (length <= 2 * this._rangeChunkSize) {
-      // The file size is smaller than the size of two chunks, so it does
-      // not make any sense to abort the request and retry with a range
-      // request.
-      return false;
-    }
-
-    return true;
+    return fullRequestXhr.getResponseHeader(name);
   },
 
   _onHeadersReceived:
       function PDFNetworkStreamFullRequestReader_onHeadersReceived() {
-
-    if (this._validateRangeRequestCapabilities()) {
+    // Setting right content length.
+    this._contentLength =
+      parseInt(this.getResponseHeader('Content-Length'), 10);
```

Comment:
> `this._contentLength` will be `NaN` which probably isn't very good (compare with the old code which uses a !isInt(...) check)?

Yeah, you are right. Sorry I didn't noticed that. I will add check for `isInt`. Thanks.

>  you're always setting this._contentLength whereas before that was only done when range requests were supported

This commit is not intended to change any functionality of the network.js file(just to extract the logic for range request support). I think setting `this._contentLength` inside `if (validateRangeRequestCapability) {...` would be better. Thanks for pointing out.